frontend/components/Leaderboard.tsx

```diff
@@ -35,49 +36,112 @@ const Leaderboard: React.FC = () => {
   );
 
   useEffect(() => {
+    let isMounted = true;
+    const delay = (ms: number) =>
+      new Promise((resolve) => setTimeout(resolve, ms));
+
+    const fetchProfileDataWithDelay = async (
+      contributors: string[],
+      delayMs: number,
+    ) => {
+      const profiles = [];
+      for (const contrib of contributors) {
+        if (!isMounted) return;
+        const profile = await fetchProfileData(contrib, networkConfig.rpcUrl);
+        profiles.push(profile);
+        await delay(delayMs);
+      }
+      return profiles;
+    };
+
     const fetchContributions = async () => {
       try {
         const [topContributors, contributions] =
           await burntPixArchives.getTopTenContributors();
-        if (Number(contributions[0]) === 0) {
+        if (Number(contributions[0]) === 0 || !isMounted) {
           return;
         }
+        const delayMsTime = 150;
+
+        const firstChunk = topContributors.slice(0, 5);
+        const secondChunk = topContributors.slice(5);
+
+        const firstChunkProfiles = await fetchProfileDataWithDelay(
+          firstChunk,
+          delayMsTime,
+        );
 
-        const profiles = await Promise.all(
-          topContributors.map((contrib) =>
-            fetchProfileData(contrib, networkConfig.rpcUrl),
-          ),
+        if (!isMounted || !firstChunkProfiles) return;
```

Comment:
`firstChunkProfiles` could be `[]` which is truthy. I think you mean `firstChunkProfiles.length`.
But there's also another problem with this approach. What if you have profiles in first 5 but not the yet second 5 because there's not more than 5 profiles in leaderboard yet... Then you'll run into a weird situation where `loadingSecondFive` will stay true.

In general all of this seems way too complicated and bug prone. Instead of breaking this up into batches of 5 w/ 150 milliseconds delay another approach could be to delay 30 milliseconds after each call and removing the skeleton as soon as first profile starts loading, so leaderboard gradually populates

Reply:
that is fair, I've tested it with only 5 results and it works great.
The alternative approach could also be a good idea, the only down side could be the re-renderings due to constant changes on the contributors array